src/components/trigger-volume.js

```diff
@@ -0,0 +1,64 @@
+AFRAME.registerComponent("trigger-volume", {
+  schema: {
+    colliders: { type: "selectorAll" },
+    scope: { type: "selector" },
+    size: { type: "vec3", default: { x: 1, y: 1, z: 1 } },
+    target: { type: "string" },
+    enterComponent: { type: "string" },
+    enterProperty: { type: "string" },
+    enterValue: {
+      default: "",
+      parse: v => (typeof v === "object" ? v : JSON.parse(v)),
+      stringify: JSON.stringify
+    },
+    leaveComponent: { type: "string" },
+    leaveProperty: { type: "string" },
+    leaveValue: {
+      default: "",
+      parse: v => (typeof v === "object" ? v : JSON.parse(v)),
+      stringify: JSON.stringify
+    }
+  },
+  init() {
+    this.size = new THREE.Vector3();
+    this.boundingBoxWorldPosition = new THREE.Vector3();
+    this.boundingBox = new THREE.Box3();
+    this.colliderWorldPosition = new THREE.Vector3();
+    this.collidingLastFrame = {};
+  },
+  update() {
+    this.el.object3D.getWorldPosition(this.boundingBoxWorldPosition);
+    this.size.copy(this.data.size);
+    this.boundingBox.setFromCenterAndSize(this.boundingBoxWorldPosition, this.size);
+  },
+  tick() {
+    const colliders = this.data.colliders;
+
+    this.el.object3D.getWorldPosition(this.boundingBoxWorldPosition);
+
+    for (let i = 0; i < colliders.length; i++) {
+      const collider = colliders[i];
+      const object3D = collider.object3D;
+
+      object3D.getWorldPosition(this.colliderWorldPosition);
+      const isColliding = this.boundingBox.containsPoint(this.colliderWorldPosition);
```

Comment:
Hmm. Seems pretty not ideal to just be checking if the center point of the "collider" is in the bounding box here to determine collisions. Especially since we seem to be using the whole "player rig" as the default "collider". This means in VR you have to move the center of your playspace into the trigger volume to trigger it. Could also get pretty weird with fly mode. Seems like we should be using the physics system for this, since it already handles all this and even has collision enter/exit events (see `sticky-object-zone` https://github.com/mozilla/hubs/blob/master/src/components/sticky-object.js#L104)

Reply:
Oh yeah, this is going to break if I don't use the avatar position. Good catch 👍

Is the physics system really better here? I think checking for a single point inside a box collider is just about as fast as you can get.

Reply:
I think the only entity that I can use as the collider is the `player-camera`. `AvatarRoot` may not be there right?

Reply:
Yeah, not suggesting physics here as a perf optimization but as a UX optimization. It might feel weird to have to get the center of my head inside the volume, but that might just be a matter of getting the volumes sized correctly. That said I think we are going to be computing these sorts of collisions anyway, so not sure it would be super expensive anyway.

Reply:
I'm not confident enough that moving to the physics system would provide much of a UX benefit. The scene creator will have to size their trigger volumes to account for the user's head, but that doesn't seem like a deal breaker. On the other hand, moving to the physics system will require physics a physics body on the avatar which I'm not certain I want to touch given that the new physics system is landing soon.

I think this whole feature needs another pass and a long incubation time before being exposed to users in Spoke. Can we ship with the simpler, more performant solution for the tutorial space and revisit in a few weeks after the physics system has landed?